### PR TITLE
update CI for onnxruntime-web

### PR DESCRIPTION
**Pull request by fs-eire**
**Description**: This change extends the existing CI for WebAssembly and enable it for onnxruntime-web.

The pipeline contains 2 jobs:
 - build_WASM: `--build_wasm`. contains 4 build matrix.
 - build_onnxruntime_web: consume the webassembly artifacts and use it in NPM package onnxruntime-web, build and test in Chrome browser.

### CODE REVIEW

**Review by duli2012 — DISMISSED** [review]

File: cmake/CMakeLists.txt [hunk 1]
```diff
@@ -264,9 +265,19 @@ if (NOT MSVC AND NOT onnxruntime_ENABLE_BITCODE)
   string(APPEND CMAKE_C_FLAGS " -ffunction-sections -fdata-sections")
 endif()
 
-# Build WebAssembly source map on debug build.
 if (onnxruntime_BUILD_WEBASSEMBLY)
-  string(APPEND CMAKE_CXX_FLAGS_DEBUG " -g4 --source-map-base http://localhost:8000/")
+  # Enable LTO for release build
+  if (NOT CMAKE_BUILD_TYPE STREQUAL "Debug")
+    string(APPEND CMAKE_CXX_FLAGS " -flto")
```
Comment (snnn) [inline cmake/CMakeLists.txt:271]: But we already have a cmake option for this?

Comment (fs-eire) [reply cmake/CMakeLists.txt:271]: yes, but it looks like `set (CMAKE_INTERPROCEDURAL_OPTIMIZATION ON)` does not work correctly on emcc.

do you suggest to append flags at https://github.com/microsoft/onnxruntime/blob/73a6af52db6fd14c7e0de918ba12618fb582a5d0/cmake/CMakeLists.txt#L341

Comment (fs-eire) [reply cmake/CMakeLists.txt:271]: I think I can utilize this flag. i am doing some local testing and will update this part once done

Comment (fs-eire) [reply cmake/CMakeLists.txt:274]: after some tests, I decide do not use this flag, because it sets "-flto=thin", which is not supported by wasm-ld.

**Review by duli2012 — APPROVED** [review]

**Review by snnn — APPROVED** [review]

**Pull request merged**